components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQInjectHandler.java

```diff
@@ -47,10 +49,20 @@
     private String onErrorSeq;
     private boolean sequential;
     private SynapseEnvironment synapseEnvironment;
+    private SequenceMediator seq;
 
     public RabbitMQInjectHandler(String injectingSeq, String onErrorSeq, boolean sequential,
-                                 SynapseEnvironment synapseEnvironment) {
+                                 SynapseEnvironment synapseEnvironment) throws SynapseException {
         this.injectingSeq = injectingSeq;
+        if (injectingSeq == null || injectingSeq.equals("")) {
+            String msg = "Injecting Sequence name is not specified.";
+            log.error(msg);
+            throw new SynapseException(msg);
+        }
+        seq = (SequenceMediator) synapseEnvironment.getSynapseConfiguration().getSequence(injectingSeq);
+        if (seq != null && !seq.isInitialized()) {
```

Comment:
Let's throw a SynapseExcpetion if the sequence is null

Reply:
Fixed in d926aaf94be8b06f295411ec90556f84d278cf52